codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -199,6 +199,12 @@ update these file paths accordingly.
     required String parentVirtualPath,
   }) async {
     log.info('Visiting directory ${directory.absolute.path}');
+    final String symlinkOrFilename = await symlink(directory.absolute.path, processManager);
+    if (symlinkOrFilename != directory.absolute.path) {
+      log.info("current direcotry is a symlink to $symlinkOrFilename, "
+          "codesign is therefore skipped for the current directory");
```

Comment:
File or directory?

Reply:
umm I implemented it as a check on directory path
for example:
```
xilaizhang-macbookpro:~ xilaizhang$ readlink -f Downloads/FlutterMacOS.framework/Versions/Current
/Users/xilaizhang/Downloads/FlutterMacOS.framework/Versions/A
xilaizhang-macbookpro:~ xilaizhang$ readlink -f Downloads/FlutterMacOS.framework/Versions
/Users/xilaizhang/Downloads/FlutterMacOS.framework/Versions
```
the directory which starts to become a symlink will return a different path when `readlink` is invoked

Reply:
I believe the same logic works for files. In the future we may have both directory and file symlinks

Reply:
Ah true, yeah sir Godofredo is right. it works for both directory and files. Umm would we have a scenario where /directory is not a symlink but /directory/file is a symlink?

If so, I need to update the logic to process the files as well. I will work on symlink files logic once I arrive at office tomorrow. thanks for the catch!

Reply:
No sure if we need it now but it is probably easier to fix that now rather than spending cycles later to triage the problem when we introduce file symlinks

Reply:
we should not be calling readlink, but instead using `dart:io`'s APIs. Let me write up some sample code...

Reply:
ahh but reading more, I see the other problem is that we need to be more robust about what entities we check

Reply:
I'll follow up with a PR later to migrate this off of readlink, but that is unrelated to this discussion.

Reply:
thanks for the insights! just to verify, umm does this mean it is okay to use readlink for now? I am halfway through getting it done with readlink

Reply:
Updated the logic to check symlink in visitBinary. verified locally and get the expected output
```
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignZIT6l1/single_artifact/FlutterMacOS.framework/Versions/Current/FlutterMacOS is a symlink to /private/var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignZIT6l1/single_artifact/FlutterMacOS.framework/Versions/A/FlutterMacOS
...
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignZIT6l1/single_artifact/FlutterMacOS.framework/FlutterMacOS is a symlink to /private/var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignZIT6l1/single_artifact/FlutterMacOS.framework/Versions/A/FlutterMacOS
, codesign is therefore skipped for the current file
```

Reply:
Actually there is something wrong with the current logic. it skips the non-symlink binary as well, inspecting. 
```
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignZIT6l1/single_artifact/FlutterMacOS.framework/Versions/A/FlutterMacOS is a symlink to /private/var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignZIT6l1/single_artifact/FlutterMacOS.framework/Versions/A/FlutterMacOS
```

Reply:
Looks like the system temp directory is a symlink itself, (mapping from /var/folders -> /private/var/folders), thinking how to resolve this

Reply:
I'm seeing a newline at the end of the filename, I think you should trim the stdout in the `symlink()` function (although amazingly this still works?)

Reply:
Replaced `readlink` with [list](https://api.dart.dev/stable/2.19.6/dart-io/Directory-class.html) (`followLinks: false`) from the `dart:io` family. The [Link](https://api.dart.dev/stable/2.19.6/dart-io/Link-class.html) is an implementation of [filesystem entity](https://api.dart.dev/stable/2.19.6/dart-io/FileSystemEntity-class.html) so we could check it with [isLink](https://api.dart.dev/stable/2.19.6/dart-io/FileSystemEntity/isLink.html).

Verified locally that the output are correct and expected
```
[INFO] codesign: signing file at path /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignEPQEqT/single_artifact/FlutterMacOS.framework/Versions/A/FlutterMacOS
...
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignEPQEqT/single_artifact/FlutterMacOS.framework/Resources is a symlink to Versions/Current/Resources, codesign is therefore skipped for the current file
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignEPQEqT/single_artifact/FlutterMacOS.framework/Versions/Current is a symlink to A, codesign is therefore skipped for the current file
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignEPQEqT/single_artifact/FlutterMacOS.framework/Headers is a symlink to Versions/Current/Headers, codesign is therefore skipped for the current file
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignEPQEqT/single_artifact/FlutterMacOS.framework/Modules is a symlink to Versions/Current/Modules, codesign is therefore skipped for the current file
[INFO] codesign: current file /var/folders/2p/d9w0246x2kd8_49q4xs4055c00xkgf/T/conductor_codesignEPQEqT/single_artifact/FlutterMacOS.framework/FlutterMacOS is a symlink to Versions/Current/FlutterMacOS, codesign is therefore skipped for the current file
```

This approach should resolve all the previously mentioned problems.